code/datums/status_effects/debuffs.dm

```diff
@@ -180,31 +180,69 @@
 	ADD_TRAIT(owner, TRAIT_KNOCKEDOUT, TRAIT_STATUS_EFFECT(id))
 	tick_interval = initial(tick_interval)
 
+#define HEALING_SLEEP_DEFAULT -0.2
+
 /datum/status_effect/incapacitating/sleeping/tick()
 	if(owner.maxHealth)
 		var/health_ratio = owner.health / owner.maxHealth
-		var/healing = -0.2
+		var/healing = HEALING_SLEEP_DEFAULT
+
+		// having high spirits helps us recover
+		var/datum/component/mood/mood = owner.GetComponent(/datum/component/mood)
+		if(mood != null)
+			switch(mood.sanity_level)
+				if(SANITY_LEVEL_GREAT)
+					healing = -0.2
+				if(SANITY_LEVEL_NEUTRAL)
+					healing = -0.1
+				if(SANITY_LEVEL_DISTURBED)
+					healing = 0
+				if(SANITY_LEVEL_UNSTABLE)
+					healing = 0
+				if(SANITY_LEVEL_CRAZY)
+					healing = 0.1
+				if(SANITY_LEVEL_INSANE)
+					healing = 0.2
+
+		var/turf/rest_turf = get_turf(owner)
+		var/is_sleeping_in_darkness = rest_turf.get_lumcount() <= 0.2 // switch 0.2 to the LIGHTING_IS_DARK define when Paper DLC gets merged
```

Comment:
why the fuck is the paper dlc pr adding light level defines? reading?

Reply:
Exactly.  The plan is to later on make it something that is usable by the `canUseTopic` proc as a bitflag. (so people can't do stuff like surgery in the dark, etc.)

Reply:
- #66308 Was merged so I can finally use the proper define with https://github.com/tgstation/tgstation/pull/65713/commits/415b8408e3bcb35c40098d5925adef711236e8d3